users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -0,0 +1,300 @@
+from typing import Dict, Union, Optional, List
+
+
+def add_specaug_layer(
+    network: Dict,
+    name: str = "specaug",
+    from_list: Optional[Union[str, List[str]]] = None,
+    num_epochs: int = 450,
+    config: Dict = None,
+) -> tuple:
+    """
+    Add a customizable SpecAugment layer to the network.
+
+    This function adds a highly configurable SpecAugment layer to the given network,
+    allowing for fine-grained control over various aspects of the augmentation process.
+
+    Args:
+        network (Dict): The network to which the SpecAugment layer will be added.
+        name (str, optional): The name of the SpecAugment layer. Defaults to "specaug".
+        from_list (Optional[Union[str, List[str]]], optional): The input layer(s) for the SpecAugment layer.
+            Default is ["data"].
+        config (Dict, optional): A dictionary containing configuration options for the SpecAugment layer.
+            Config Dict Parameters:
+                max_time_num (int): The beginning maximum number of time masks to be applied. Default is 3.
+                max_time (int): The beginning maximum size (in time frames) of each time mask. Default is 10.
+                max_feature_num (int): The beginning maximum number of frequency masks to be applied. Default is 4.
+                max_feature (int): The beginning maximum size (in frequency bins) of each frequency mask. Default is 5.
+                enable_sorting (bool): Whether to sort filters by their center frequency before applying masks. Default is True.
+                steps_per_epoch (int): The number of steps per epoch. Make sure this parameter is accurate since the all the scheduling depends on it. Default is 2080.
+                sorting_start_epoch (int): The subepoch number to start sorting filters. Default is 0.
+                mask_growth_strategy (str): The strategy for increasing the mask sizes over epochs (linear or step). Default is "linear".
+                time_mask_num_schedule (Dict[int, float]): A dictionary mapping subepoch numbers to the multiplicator for the time mask number.
+                    Default is {0: 1, 10: 1.5, 20: 2}.
+                time_mask_size_schedule (Dict[int, float]): A dictionary mapping subepoch numbers to the multiplicator for the time mask size.
+                    Default is {0: 1, 15: 1.5, 25: 2}.
+                freq_mask_num_schedule (Dict[int, float]): A dictionary mapping subepoch numbers to the multiplicator for the frequency mask number.
+                    Default is {0: 1, 5: 1.2, 15: 1.5}.
+                freq_mask_size_schedule (Dict[int, float]): A dictionary mapping subepoch numbers to the multiplicator for the frequency mask size.
+                    Default is {0: 1, 10: 1.2, 20: 1.5}.
```

Comment:
I would not set such defaults. They are highly tuning-dependent. They should be in the `experiments.py` file, not here. The baseline behaviour should be the regular SpecAugment behaviour imho.